openedx_tagging/core/tagging/rest_api/v1/views.py

```diff
@@ -556,3 +634,81 @@ def get_queryset(self) -> list[Tag]:  # type: ignore[override]
         self.pagination_class = self.get_pagination_class()
 
         return result
+
+    def post(self, request, *args, **kwargs):
+        """
+        Creates new Tag in Taxonomy and returns the newly created Tag.
+        """
+        pk = self.kwargs.get("pk")
+        taxonomy = self.get_taxonomy(pk)
+
+        body = TaxonomyTagCreateBodySerializer(data=request.data)
+        body.is_valid(raise_exception=True)
+
+        tag = body.data.get("tag")
+        parent_tag_id = body.data.get("parent_tag_id", None)
+        external_id = body.data.get("external_id", None)
+
+        try:
+            new_tag = add_tag_to_taxonomy(
+                taxonomy, tag, parent_tag_id, external_id
+            )
+        except Tag.DoesNotExist as e:
```

Comment:
I just noticed that the [api exports `TagDoesNotExist = Tag.DoesNotExist`](https://github.com/openedx/openedx-learning/blob/d44775d83e00e642c96077da93d6ed82c06204be/openedx_tagging/core/tagging/api.py#L22), so maybe we should use that instead?

Reply:
Sure thing, updated